### DISCUSSION TITLE
how to get the parameter values from material instance or material?

### DISCUSSION THREAD

**Discussion by zhangbo-0213** [google/filament#5647] [Q&A]
how to get the parameter values from materialInstance or material, that I could reset these value after I changed parameter value in material ?

**Comment by zhangbo-0213** [comment]
is there any  api like this:   
material.GetFloat("parameterName"); 
material.GetColor("parameterName");   
material.GetTexture("parameterName");
in filament?

**Reply by bejado** [reply]
No, we don't have any API for this. It's up to the app to keep track of this.

**Reply by zhangbo-0213** [reply]
Thanks for reply ,so how can I track this parameter value of material Instance which from the GLTF asset ?

**Reply by bejado** [reply]
I can't think of any easy way to do this with gltfio. Maybe @prideout has some ideas.

**Reply by prideout** [reply]
If you're using C++ you can look at the raw glTF data via getSourceAsset.

**Reply by zhangbo-0213** [reply]
OK，thanks
